pkg/apis/eksctl.io/v1alpha4/defaults.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha4
+
+// SetNodeGroupDefaults will set defaults for a given nodegroup
+func SetNodeGroupDefaults(_ int, ng *NodeGroup) error {
+	if ng.InstanceType == "" {
+		ng.InstanceType = DefaultNodeType
+	}
+	if ng.AMIFamily == "" {
+		ng.AMIFamily = DefaultNodeImageFamily
+	}
+	if ng.AMI == "" {
+		ng.AMI = "static"
+	}
+
+	if ng.SecurityGroups == nil {
+		ng.SecurityGroups = &NodeGroupSGs{
+			AttachIDs: []string{},
+		}
+	}
+	if ng.SecurityGroups.WithLocal == nil {
+		ng.SecurityGroups.WithLocal = NewBoolTrue()
+	}
+	if ng.SecurityGroups.WithShared == nil {
+		ng.SecurityGroups.WithShared = NewBoolTrue()
+	}
+
+	// Enable SSH when a key is provided
+	if ng.SSHPublicKeyPath != "" && ng.SSHPublicKeyPath != DefaultNodeSSHPublicKeyPath {
```

Comment:
@martina-if this is a temporary fix, I think we should be able to get rid of `!= DefaultNodeSSHPublicKeyPath` when we switch to references.